Themes showcase: remove Beta badge for FSE themes

creativecoder opened the pull request:
#### Changes proposed in this Pull Request

* Remove the "Beta" badge for block themes on FSE eligible sites
* Remove the "Full Site Editing" theme tab, since the themes are the same as what we show in the Recommended tab

Now that we are rapidly rolling out FSE to new users through the /start flow without any messaging about it being in beta, it seems better not to have "Beta" plastered all over the themes showcase.

#### Testing instructions

* On an FSE eligible site (you can create one through the wordpress.com/new flow, or horizon.wordpress.com/start)
* See that the themes showcase no longer shows the Beta badge on FSE themes
* And that the "Full Site Editing" theme tab is gone.

| **Before** | **After** |
| - | - |
| <img width="718" alt="image" src="https://user-images.githubusercontent.com/1699996/151635716-dab8efeb-d129-4db9-848d-183238e0ff1e.png"> | <img width="702" alt="image" src="https://user-images.githubusercontent.com/1699996/151635746-bedcb6e3-9f54-4ba1-a085-6f591c9557e6.png"> |



Resolves https://github.com/Automattic/wp-calypso/issues/58930

Commit:
Themes showcase: remove Beta badge for FSE themes

zaguiini commented on client/components/theme/index.jsx:
```diff
@@ -145,7 +143,7 @@ export class Theme extends Component {
 	};
 
 	render() {
-		const { active, blockEditorSettings, price, theme, translate, upsellUrl } = this.props;
```
Now that we're not using `blockEditorSettings` anymore, we can remove the HOC below and the prop type declaration.

jeyip replied:
Done 👍

zaguiini commented on client/my-sites/themes/theme-showcase.jsx:
```diff
@@ -284,9 +270,6 @@ class ThemeShowcase extends Component {
 				return this.props.isJetpackSite;
-			case this.tabFilters.FSE.key:
-				// Display FSE tab if the Site Editor is active for the site.
-				return this.props.blockEditorSettings?.is_fse_eligible;
```
The same thing here, we can get rid of `blockEditorSettings` for this file.

zaguiini reviewed (commented):
There's also:

- `client/my-sites/theme/main.jsx`;
- `client/my-sites/themes/current-theme/index.tsx`...

displaying the `Beta` badge. Will we get rid of it there as well?

We probably can un-HOC `client/my-sites/themes/theme-showcase.jsx` too because the prop is not being used there anymore, nor in the children that gets props from this component.

jeyip committed:
Remove references to beta badges in the current theme

jeyip committed:
Remove block editor settings HOC from theme component

jeyip committed:
Remove beta badge from theme info page

jeyip commented:
@creativecoder I made the changes that @zaguiini suggested. Could you do another smoke test? Things look good on my end.

jeyip commented:
## Testing
### Requirements
- [x] The themes cards in the themes showcase no longer show the Beta badge on FSE themes
- [x] The current theme card in the themes showcase no longer shows the Beta badge for an FSE theme
<img width="760" alt="Screen Shot 2022-02-01 at 10 51 39 AM" src="https://user-images.githubusercontent.com/5414230/152032568-d7ef9dec-2701-48af-b90b-59a6397755a0.png">
- [x] The theme info page no longer shows a beta badge for an FSE theme
<img width="1504" alt="Screen Shot 2022-02-01 at 10 52 04 AM" src="https://user-images.githubusercontent.com/5414230/152032521-02eff5e4-3c60-4847-84d5-18194f98ef0e.png">
- [x]  The "Full Site Editing" theme tab is gone.
### Browsers
- [x] Edge
- [x] Firefox
- [x] Safari
- [x] Chrome

Commit:
Remove unused proptype

creativecoder commented:
Sorry about the mass ping, y'all--I borked a rebase.

Commit:
Merge branch 'trunk' into remove/theme-beta-badge

Commit:
Update themes-list snapshot

Commit:
Disconnect blockEditorSettings from theme showcase

creativecoder commented:
Thanks for the suggestions @zaguiini and for the changes @jeyip . I've also removed the use of another blockEditorSettings HOC, deleted a couple of now unused prop types, and fixed a test. This is ready for a final review.

jeyip reviewed (approved):
LGTM 🚢
